libraries/bezier-rs/src/subpath/lookup.rs

```diff
@@ -7,13 +9,34 @@ impl Subpath {
 	pub fn length(&self, num_subdivisions: Option<usize>) -> f64 {
 		self.iter().fold(0., |accumulator, bezier| accumulator + bezier.length(num_subdivisions))
 	}
+
+	/// Returns the segment index and `t` value that corresponds to the closest point on the curve to the provided point.
+	/// Uses a searching algorithm akin to binary search that can be customized using the [ProjectionOptions] structure.
+	pub fn project(&self, point: DVec2, options: ProjectionOptions) -> (usize, f64) {
+		if self.is_empty() {
+			panic!("Can not project to an empty Subpath")
```

Comment:
@0HyperCube @TrueDoctor thoughts on whether this should return `Option`, `Result`, or panic?

Reply:
I'd favor returning an Option because empty is afaict a valid state for Subpath.